arduino/KnexElevator/ITask.cpp

```diff
@@ -0,0 +1,101 @@
+//
+// MIT License
+//
+// Copyright (c) 2017 gfarrand (grant.farrand@gmail.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+// /////////////////////////////////////////////////////////////////////////////
+
+
+// /////////////////////////////////////////////////////////////////////////////
+// Include(s)
+// /////////////////////////////////////////////////////////////////////////////
+#include <Arduino_FreeRTOS.h>
+#include <Arduino.h>
+
+#include "ITask.h"
+
+
+////////////////////////////////////////////////////////////////////////////////
+ITask::ITask(char const * taskName, int taskPriority, int stackDepth)
+    : mTaskName()
+    , mTaskPriority(taskPriority)
+    , mTaskStackDepth(stackDepth)
+    , mTaskHandle(NULL)
+{
+    strncpy(mTaskName, taskName, sizeof(mTaskName));
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+ITask::~ITask(void)
+{
+    // Intentionally left empty
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+void
+ITask::Start(void)
+{
+    Serial.print("Now starting task '");
+    Serial.print(mTaskName);
+    Serial.println("'");
+
+    // Fire up our task
+    xTaskCreate(ITask::RunWrapper,
+                mTaskName,
+                mTaskStackDepth,
+                this,
+                mTaskPriority,
+                &mTaskHandle);
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+void
+ITask::Stop(void)
+{
+    Serial.print("Now stopping task '");
+    Serial.print(mTaskName);
+    Serial.println("'");
+
+    // Remove our task from execution
+    vTaskDelete(mTaskHandle);
+    mTaskHandle = NULL;
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+void
+ITask::RunWrapper(void * taskParameters)
+{
+    // Entry point to the new task
+    ITask* thisTask =
```

Comment:
@saunmeht - There is still no checking for NULL here, because the Start() method is now the only place in code where a NULL pointer could be passed, and that is being given the value `this`. Perhaps this is sufficient to verify by inspection?